soufi/finders/yum.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) 2021 Cisco Systems, Inc. and its affiliates
+# All rights reserved.
+
+import functools
+import urllib
+
+import repomd
+import requests
+
+from soufi import exceptions, finder
+
+TIMEOUT = 30  # seconds
+
+
+class YumFinder(finder.SourceFinder):
+    """A base class for Yum-based repositories.
+
+    Subclasses of this should provide methods for setting up the repository
+    search lists to pass to __init__.
+
+    The lookup is a 2-stage process:
+        1. Attempt to look up the SRPM directly from the sources repository.
+           In these cases, we use the name only and ignore the version,
+           since it is a very common practice for repositories to only
+           publish repodata for the "current" version, while still
+           providing older sources.
+        2. Attempt to look up the name of the source RPM from the binary
+           repository.  This will catch packages where the source and
+           binary package names do not match.  The version is also ignored
+           in this step, for the same reasons.  Then backtrack and attempt
+           to lookup that package in the source repos.
+    """
+
+    distro = None
```

Comment:
I realize this may not be the correct behavior, but this is a fully-capable finder on its own, provided that it is initialized with a valid set of search paths.  It's just that we're nominally discouraging that in favor of loading the derived finders.

It **DOES** seems to work like this, though.  Suggestions for improvement welcome.